File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/AuthorForm.ts
import { StringEditor, EnumEditor, EmailAddressEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { SexEnum } from "../Modules/Common.Enums.SexEnum";

export interface AuthorForm {
    AuthorName: StringEditor;
    Sex: EnumEditor;
    Email: EmailAddressEditor;
    AuthorImage: ImageUploadEditor;
}

export class AuthorForm extends PrefixedContext {
    static readonly formKey = 'BookManage.Author';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AuthorForm.init)  {
            AuthorForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = EmailAddressEditor;
            var w3 = ImageUploadEditor;

            initFormType(AuthorForm, [
                'AuthorName', w0,
                'Sex', w1,
                'Email', w2,
                'AuthorImage', w3
            ]);
        }
    }
}

[SexEnum]; // referenced types

File: LMIS/LMIS.Web/Modules/BookManage/Author/AuthorForm.cs
using LMIS.Modules.Common.Enums;
using Serenity.ComponentModel;
using System;

namespace LMIS.BookManage.Forms;

[FormScript("BookManage.Author")]
[BasedOnRow(typeof(AuthorRow), CheckNames = true)]
public class AuthorForm
{
    public string AuthorName { get; set; }
    public SexEnum Sex { get; set; }
    [EmailAddressEditor]
    public string Email { get; set; }
    public string AuthorImage { get; set; }
}